src/metabase/types.clj

```diff
@@ -5,6 +5,7 @@
 (derive :type/Dictionary :type/Collection)
 (derive :type/Array :type/Collection)
 
+(derive :type/Nil :type/*)
```

Comment:
sure. But you didn't make that change everywhere else in the codebase so now we have two ways for a `special_type` to be `nil`: either `nil` (everywhere else) or `:type/Nil` (only in one single place).